src/game/utils/flop_helper.rs

```diff
@@ -0,0 +1,261 @@
+        "TcTdTh", "JhTcTd", "QhTcTd", "KhTcTd", "AhTcTd", "JdJhTc", "QhJdTc", "KhJdTc", "AhJdTc",
+        "QdQhTc", "KhQdTc", "AhQdTc", "KdKhTc", "AhKdTc", "AdAhTc", "KcQcJc", "AcQcJc", "QcJc2d",
+        "QcJc3d", "QcJc4d", "QcJc5d", "QcJc6d", "QcJc7d", "QcJc8d", "QcJc9d", "QcJcTd", "QcJcJd",
+        "QcQdJc", "KdQcJc", "AdQcJc", "AcKcJc", "KcJc2d", "KcJc3d", "KcJc4d", "KcJc5d", "KcJc6d",
+        "KcJc7d", "KcJc8d", "KcJc9d", "KcJcTd", "KcJcJd", "KcQdJc", "KcKdJc", "AdKcJc", "AcJc2d",
+        "AcJc3d", "AcJc4d", "AcJc5d", "AcJc6d", "AcJc7d", "AcJc8d", "AcJc9d", "AcJcTd", "AcJcJd",
+        "AcQdJc", "AcKdJc", "AcAdJc", "JcJdJh", "QhJcJd", "KhJcJd", "AhJcJd", "QdQhJc", "KhQdJc",
+        "AhQdJc", "KdKhJc", "AhKdJc", "AdAhJc", "AcKcQc", "KcQc2d", "KcQc3d", "KcQc4d", "KcQc5d",
+        "KcQc6d", "KcQc7d", "KcQc8d", "KcQc9d", "KcQcTd", "KcQcJd", "KcQcQd", "KcKdQc", "AdKcQc",
+        "AcQc2d", "AcQc3d", "AcQc4d", "AcQc5d", "AcQc6d", "AcQc7d", "AcQc8d", "AcQc9d", "AcQcTd",
+        "AcQcJd", "AcQcQd", "AcKdQc", "AcAdQc", "QcQdQh", "KhQcQd", "AhQcQd", "KdKhQc", "AhKdQc",
+        "AdAhQc", "AcKc2d", "AcKc3d", "AcKc4d", "AcKc5d", "AcKc6d", "AcKc7d", "AcKc8d", "AcKc9d",
+        "AcKcTd", "AcKcJd", "AcKcQd", "AcKcKd", "AcAdKc", "KcKdKh", "AhKcKd", "AdAhKc", "AcAdAh",
+    ]
+    .iter()
+    .map(|flop_str| flop_from_str(flop_str).unwrap())
+    .collect();
+}
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum Texture {
+    Monotone,
+    Twotone,
+    Rainbow,
+    Paired,
+    Triple,
+    Unpaired,
+}
+
+// Filters all flops by the high card, ignoring suit
+pub fn high_flops(card: Card) -> Vec<[Card; 3]> {
+    high_flops_from_list(card, ALL_FLOPS.iter().copied().collect())
+}
+
+pub fn high_flops_from_list(card: Card, flop_list: Vec<[Card; 3]>) -> Vec<[Card; 3]> {
+    flop_list
+        .iter()
+        .copied()
+        .filter(|flop| flop[2] >> 2 == card >> 2)
+        .collect()
+}
+
+pub fn textured_flops(texture: Texture) -> Vec<[Card; 3]> {
+    textured_flops_from_list(texture, ALL_FLOPS.iter().copied().collect())
+}
+
+pub fn textured_flops_from_list(texture: Texture, flop_list: Vec<[Card; 3]>) -> Vec<[Card; 3]> {
+    flop_list
+        .iter()
+        .copied()
+        .filter(|flop| {
+            let suits: HashSet<Card> = HashSet::from_iter(flop.iter().copied().map(|c| c % 4));
+            let values: HashSet<Card> = HashSet::from_iter(flop.iter().copied().map(|c| c >> 2));
+            match texture {
+                Texture::Monotone => suits.len() == 1,
+                Texture::Twotone => suits.len() == 2,
+                Texture::Rainbow => suits.len() == 3,
+                Texture::Triple => values.len() == 1,
+                Texture::Paired => values.len() == 2,
+                Texture::Unpaired => values.len() == 3,
+            }
+        })
+        .collect()
+}
+
+pub fn flop_to_string(flop: &[u8; 3]) -> String {
```

Comment:
i think this might be somewhere else already?

Reply:
You might be thinking of `flop_from_str`, which goes in the opposite direction.

Reply:
Could move this into `src/range.rs`, where `flop_from_str` is defined.

Reply:
Yeah maybe, I could have sworn I'd seen it somewhere but you're probably right. I might have even written it in another branch for serialization?

Anyway, I think `range` is a good spot for it for now, but I think I have a separate issue about factoring that stuff out of `range`. The layout of this stuff is not great 🙃

Reply:
moved to range.